localstack-core/localstack/utils/lambda_debug_mode/lambda_debug_mode_config.py

```diff
@@ -0,0 +1,198 @@
+            key = loader.construct_object(key_node, deep=deep)
+            if key in mapping:
+                # Create a MarkedYAMLError to indicate the duplicate key issue
+                raise MarkedYAMLError(
+                    context="while constructing a mapping",
+                    context_mark=node.start_mark,
+                    problem=f"found duplicate key: {key}",
+                    problem_mark=key_node.start_mark,
+                )
+            value = loader.construct_object(value_node, deep=deep)
+            mapping[key] = value
+        return mapping
+
+
+def from_yaml_string(yaml_string: str) -> Optional[LambdaDebugModeConfig]:
+    try:
+        data = yaml.load(yaml_string, _SafeLoaderWithDuplicateCheck)
+    except yaml.YAMLError as yaml_error:
+        LOG.error(
+            f"Could not parse yaml lambda debug mode configuration file due to: {str(yaml_error)}"
+        )
+        data = None
+    if not data:
+        return None
+    config = LambdaDebugModeConfig(**data)
+    return config
+
+
+def validate_lambda_debug_mode_config(config: LambdaDebugModeConfig) -> None:
+    _validate_lambda_debug_mode_config(
+        validation_state=_LambdaDebugModeConfigValidationState(), config=config
+    )
+
+
+def _validate_lambda_debug_mode_config(
+    validation_state: _LambdaDebugModeConfigValidationState, config: LambdaDebugModeConfig
+):
+    config_functions = config.functions
+    lambda_arns = list(config_functions.keys())
+    for lambda_arn in lambda_arns:
+        qualified_lambda_arn = _to_qualified_lambda_function_arn(lambda_arn)
+        if lambda_arn != qualified_lambda_arn:
+            if qualified_lambda_arn in config_functions:
+                raise DuplicateLambdaDebugConfig(
+                    lambda_arn_debug_config_first=lambda_arn,
+                    lambda_arn_debug_config_second=qualified_lambda_arn,
+                )
+            config_functions[qualified_lambda_arn] = config_functions.pop(lambda_arn)
+
+    for lambda_arn, lambda_debug_config in config_functions.items():
+        _validate_lambda_debug_config(
+            validation_state=validation_state, lambda_debug_config=lambda_debug_config
+        )
+
+
+def _to_qualified_lambda_function_arn(lambda_arn: Arn) -> Arn:
+    # Returns the $LATEST qualified version of a structurally unqualified version of a lambda Arn iff this
+    # if detected to be structurally unqualified. Otherwise, it returns the given string.
+    if not lambda_arn:
+        return lambda_arn
+    lambda_arn_parts = lambda_arn.split(":")
+    lambda_arn_parts_len = len(lambda_arn_parts)
+
+    # The arn is qualified and with a non-empy qualifier.
+    is_qualified = lambda_arn_parts_len == 8
+    if is_qualified and lambda_arn_parts[-1]:
+        return lambda_arn
+
+    # The arn is not unqualified, but probably erroneous, pass the value upstream.
+    is_unqualified = lambda_arn_parts_len == 7
+    if not is_unqualified:
+        return lambda_arn
```

Comment:
As far as I can see, we just ignore invalid arns, for example those ending with `:` (so they have 8 parts, but are missing the qualifier, or totally invalid arns) and still set them in the internal config? Personally, I would prefer raising an error to silently accepting them, because they will never match anything.